src/pool/mempool.rs

```diff
@@ -21,95 +33,146 @@ pub type KakarotPool<Client> = Pool<Validator<Client>, TransactionOrdering, Noop
 
 #[derive(Debug, Default)]
 pub struct AccountManager {
-    accounts: HashSet<Felt>,
+    accounts: Arc<Mutex<HashMap<Felt, Felt>>>,
 }
 
 impl AccountManager {
-    pub fn new(path: &str) -> Self {
-        let mut accounts = HashSet::new();
+    pub async fn new<SP>(path: &str, eth_provider: &EthDataProvider<SP>) -> eyre::Result<Self>
+    where
+        SP: starknet::providers::Provider + Send + Sync + 'static,
+    {
+        let mut accounts = HashMap::new();
 
         // Open the file specified by `path`
         let Ok(mut file) = File::open(path) else {
-            return Self::default();
+            return Ok(Self::default());
         };
 
         let mut contents = String::new();
         if file.read_to_string(&mut contents).is_err() {
-            return Self::default();
+            return Ok(Self::default());
         }
 
         // Parse the file contents as JSON
         let json: Value = match serde_json::from_str(&contents) {
             Ok(json) => json,
             Err(_) => {
-                return Self::default();
+                return Ok(Self::default());
             }
```

Comment:
can't we propagate errors with `?`?

Reply:
Yes now that I introduced error in this function this is possible, just fixed